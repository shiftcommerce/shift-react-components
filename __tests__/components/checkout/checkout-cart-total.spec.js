// Libraries
import React from 'react'

// Components
import CheckoutCartTotal from '../../../src/components/checkout/checkout-cart-total'

test("renders the cart's total, subtotal and shipping cost", () => {
  // Act
  const wrapper = mount(
    <CheckoutCartTotal
      discountSummaries={[]}
      shippingTotal={5}
      subTotal={15}
      total={20}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Total Price:  £15.00')
  expect(wrapper).toIncludeText('Shipping costs:  £5.00')
  expect(wrapper).toIncludeText('TOTAL:  £20.00')
})

test('renders promotions', () => {
  // Act
  const wrapper = mount(
    <CheckoutCartTotal
      discountSummaries={[{
        id: 1,
        name: 'Christmas Special',
        total: 1.5
      }]}
      shippingTotal={5}
      subTotal={15}
      total={20}
    />
  )

  // Assert
  expect(wrapper).toIncludeText('Christmas Special:- £1.50')
})

test('renders shipping promotions', () => {
  // Act
  const wrapper = mount(
    <CheckoutCartTotal
      discountSummaries={[]}
      shippingDiscount={-5}
      shippingDiscountName='Free shipping'
      shippingTotal={5}
      subTotal={15}
      total={20}
    />
  )

  // Assert
  expect(wrapper).toIncludeText('Free shipping:- £5.00')
})

test('renders payment errors', () => {
  // Act
  const wrapper = mount(
    <CheckoutCartTotal
      discountSummaries={[]}
      paymentError='Payment authentication failed'
      shippingTotal={5}
      subTotal={15}
      total={20}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Payment authentication failed')
})
