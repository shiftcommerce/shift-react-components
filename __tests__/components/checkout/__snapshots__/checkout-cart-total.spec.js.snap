// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders payment errors 1`] = `
<CheckoutCartTotal
  discountSummaries={Array []}
  paymentError="Payment authentication failed"
  shippingTotal={5}
  subTotal={15}
  total={20}
>
  <Sticker>
    <div>
      <div
        aria-label="Cart total summary"
        className="u-sticky c-cart-summary"
      >
        <dl
          aria-label="Subtotal"
        >
          <dt>
             Total Price: 
          </dt>
          <dd>
             £
            15.00
             
          </dd>
        </dl>
        <dl
          aria-label="Shipping cost"
        >
          <dt>
             Shipping costs: 
          </dt>
          <dd>
             
            £5.00
             
          </dd>
        </dl>
        <dl
          aria-label="Total"
          className="u-bold"
        >
          <dt>
             TOTAL: 
          </dt>
          <dd>
             £
            20.00
             
          </dd>
        </dl>
        <dl>
          <dt
            className="c-cart-summary__VAT"
          >
            * Including VAT
          </dt>
        </dl>
        <div
          className="c-checkout-cart-total__payment-error"
        >
          Payment authentication failed
        </div>
      </div>
    </div>
  </Sticker>
</CheckoutCartTotal>
`;

exports[`renders the cart's total, subtotal and shipping cost 1`] = `
<CheckoutCartTotal
  discountSummaries={Array []}
  shippingTotal={5}
  subTotal={15}
  total={20}
>
  <Sticker>
    <div>
      <div
        aria-label="Cart total summary"
        className="u-sticky c-cart-summary"
      >
        <dl
          aria-label="Subtotal"
        >
          <dt>
             Total Price: 
          </dt>
          <dd>
             £
            15.00
             
          </dd>
        </dl>
        <dl
          aria-label="Shipping cost"
        >
          <dt>
             Shipping costs: 
          </dt>
          <dd>
             
            £5.00
             
          </dd>
        </dl>
        <dl
          aria-label="Total"
          className="u-bold"
        >
          <dt>
             TOTAL: 
          </dt>
          <dd>
             £
            20.00
             
          </dd>
        </dl>
        <dl>
          <dt
            className="c-cart-summary__VAT"
          >
            * Including VAT
          </dt>
        </dl>
      </div>
    </div>
  </Sticker>
</CheckoutCartTotal>
`;
